x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx

```diff
@@ -322,24 +322,29 @@ export function LensEditConfigurationFlyout({
 
   const runQuery = useCallback(
     async (q: AggregateQuery, abortController?: AbortController) => {
-      const attrs = await getSuggestions(
-        q,
-        startDependencies,
-        datasourceMap,
-        visualizationMap,
-        adHocDataViews,
-        setErrors,
-        abortController,
-        setDataGridAttrs
-      );
+      // do not run the suggestions if the query is the same as the previous one
+      const attrs = isEqual(q, prevQuery.current)
+        ? attributes
+        : await getSuggestions(
```

Comment:
Why not do this check before calling the `runQuery`? 
In this way you can avoid rerendering the `ESQLLangEditor` due to the change of `isVisualizationLoading` and you avoid calling the `setCurrentAttributes, setErrors, updateSuggestion`.
Or at least setting the loading/done flag from within this method and not in the `onTextLangQuerySubmit` and setting the flag only if strincly necessary.

Reply:
Correct amiguito, I was doing various experiments and then I reverted and didnt think of it more thorougly. https://github.com/elastic/kibana/pull/195196/commits/e692b80eceb5ad0b30afcd46b686e0e0f2a0a893